ca/ocsp.go

```diff
@@ -0,0 +1,125 @@
+package ca
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/jmhodges/clock"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/crypto/ocsp"
+
+	blog "github.com/letsencrypt/boulder/log"
+)
+
+// ocspLogQueue accumulates OCSP logging events and writes several of them
+// in a single log line. This reduces the number of log lines and bytes,
+// which would otherwise be quite high. As of Jan 2021 we do approximately
+// 550 rps of OCSP generation events. We can turn that into about 5.5 rps
+// of log lines if we accumulate 100 entries per line, which amounts to about
+// 3900 bytes per log line.
+// Summary of log line usage:
+// serial in hex: 36 bytes, separator characters: 2 bytes, status: 1 byte
+// If maxLogLen is 0, do not perform any accumulation or logging.
```

Comment:
And if maxLogLen is e.g. 20, less than the number of bytes for a single serial?

Reply:
Changed this by making the main loop more like a do / while; it's guaranteed to run at least once. So even if you set maxLogLen = 1, we'll log one entry per line.